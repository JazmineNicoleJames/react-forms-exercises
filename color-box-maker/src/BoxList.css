.BoxList {
    display: inline-block;
    align-items: center;
}