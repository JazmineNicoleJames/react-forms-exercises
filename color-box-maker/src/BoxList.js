import React, {useState} from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import { v4 as uuid } from "uuid";
import "./BoxList.css"

const BoxList = () => {

    const [box, setBox] = useState([]);
    
    
    const remove = (id) => {
        setBox(((prevBox) => prevBox.filter(box => box.id !==id)))
    }

    const addBox = (box) => {
        const id = uuid()
        let newBox = {
            ...box,
            id
        }
        setBox((prevBox) => [...prevBox, newBox]);
    }

    return (
        <div>
            <h1> Create a box!</h1>
            <NewBoxForm addBox={addBox} className="NewBoxForm" remove={remove} />
            <div className="BoxList">
                {box.map(({id, height, width, backgroundColor}) =>
                    <Box id={id} 
                        height={height}
                        width={width}
                        backgroundColor={backgroundColor}
                        remove={remove}
                    />)}
            </div>
        </div>
    )
};

export default BoxList;