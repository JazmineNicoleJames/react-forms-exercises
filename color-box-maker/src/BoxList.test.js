import BoxList from "./BoxList";
import { render, fireEvent} from "@testing-library/react";

it("renders without crashing", () => {
    render(<BoxList />)
});

it("should match snapshot", () => {
    const { asFragment } = render(<BoxList />)

    expect(asFragment()).toMatchSnapshot();
});

it("should add new box", () => {
    const { queryByText, getByLabelText, getByTestId } = render(<BoxList />);

    const widthInput = getByLabelText("Width:");
    const heightInput = getByLabelText("Height:");
    const backgroundColorInput = getByLabelText("Background color:");
    const submitBtn = queryByText("Create box!");

    fireEvent.change(widthInput, {target: {value: "100px"}});
    fireEvent.change(heightInput, {target: {value: "100px"}});
    fireEvent.change(backgroundColorInput, {target: {value: "magenta"}});
    fireEvent.click(submitBtn);

    const newBox = getByTestId('box-component');

    expect(newBox).toBeInTheDocument();
    expect(newBox).toHaveStyle("height:100px");
    expect(newBox).toHaveStyle("width:100px");
    expect(newBox).toHaveStyle("backgroundColor:magenta")
});
