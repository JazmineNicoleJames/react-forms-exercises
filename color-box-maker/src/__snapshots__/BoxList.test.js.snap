// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
       Create a box!
    </h1>
    <form
      class="NewFormBox"
      data-testid="new-box-form"
    >
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="background color"
        type="text"
        value=""
      />
      <button>
         Create box!
      </button>
    </form>
    <div
      class="BoxList"
    />
  </div>
</DocumentFragment>
`;
